### PR TITLE
MF-99: Conditions Level 2 Widget

### PR DESCRIPTION
**Pull request by denniskigen**
Screenshot 2020-01-17 at 09 35 30

### CODE REVIEW

File: src/widgets/conditions/conditions-detailed-summary.component.test.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+jest.mock("./conditions.resource", () => ({
+  performPatientConditionSearch: jest.fn()
+}));
+
+jest.mock("@openmrs/esm-api", () => ({
+  useCurrentPatient: jest.fn()
+}));
+
+const mockPatientConditionResult = {
```
Comment (jonathandick) [inline src/widgets/conditions/conditions-detailed-summary.component.test.tsx]: Could you move this mocked data into the __mocks__ directory?

File: translations/en.json [hunk 2]
```diff
@@ -1,7 +1,3 @@
 {
-  "Active Medications": "Active Medications",
-  "Active Programs": "Active Programs",
```
Comment (donaldkibet) [inline translations/en.json]: Was this deleted by mistake? we should restore the translation.

File: src/widgets/conditions/conditions-detailed-summary.component.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    }
+  }, [isLoadingPatient, patient]);
+
+  function displayConditions() {
+    return (
+      <SummaryCard
+        name="Conditions"
+        match={props.match}
+        styles={{ width: "100%" }}
+        addBtnUrl={`/patient/${patientUuid}/chart/condition/add`}
```
Comment (jonathandick) [inline src/widgets/conditions/conditions-detailed-summary.component.tsx]: I think this should be /conditions/

File: src/widgets/conditions/conditions-detailed-summary.css [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  padding: 0.25rem 0.5rem 0 0.5rem;
+}
+
+.conditionTable {
+  width: 100%;
+}
+
+.conditionTable thead {
+  color: var(--omrs-color-ink-medium-contrast);
+  border-bottom: 1px solid var(--omrs-color-bg-low-contrast);
```
Comment (donaldkibet) [inline src/widgets/conditions/conditions-detailed-summary.css]: convert this to use rem
```suggestion
  border-bottom: 0.0625rem solid var(--omrs-color-bg-low-contrast);
```

File: src/widgets/conditions/conditions-detailed-summary.css [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+:global(.omrs-breakpoint-lt-tablet) .conditionTable thead td {
+  padding: 0.25rem 0.125rem 0.5rem 0.5rem;
+}
+
+.conditionTable thead td:nth-child(2) {
+  text-align: right;
+  padding-right: 0rem;
+  background: "lightgray";
```
Comment (donaldkibet) [inline src/widgets/conditions/conditions-detailed-summary.css]:
```suggestion
  background: var(--omrs-color-ink-low-contrast);
```

File: src/widgets/conditions/conditions-detailed-summary.css [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  border-bottom: 1px solid var(--omrs-color-bg-low-contrast);
+}
+
+:global(.omrs-breakpoint-lt-tablet) .conditionTable tbody {
+  padding: 1rem;
+}
+
+.conditionTable tr {
+  padding: 13px 0 16px 13px;
+  border-bottom: 1px solid var(--omrs-color-bg-low-contrast);
```
Comment (donaldkibet) [inline src/widgets/conditions/conditions-detailed-summary.css]: same here convert to rem.

File: src/widgets/conditions/conditions-detailed-summary.component.test.tsx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+        <ConditionsDetailedSummary match={match} />
+      </BrowserRouter>
+    );
+
+    await wait(() => {
+      expect(wrapper).toBeDefined();
+    });
+  });
+
+  xit("should display the patient's conditions correctly", async () => {
```
Comment (donaldkibet) [inline src/widgets/conditions/conditions-detailed-summary.component.test.tsx]:
```suggestion
  it("should display the patient's conditions correctly", async () => {
```
unskip this test

File: src/widgets/conditions/conditions-detailed-summary.component.test.tsx [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    );
+    wrapper = render(
+      <BrowserRouter>
+        <ConditionsDetailedSummary match={match} />
+      </BrowserRouter>
+    );
+
+    await wait(() => {
+      expect(wrapper.getByText("Renal rejection").textContent).toBeTruthy();
+      // expect(wrapper.getByText("").textContent).toBeTruthy();
```
Comment (donaldkibet) [inline src/widgets/conditions/conditions-detailed-summary.component.test.tsx]:
```suggestion

```

File: src/widgets/conditions/conditions-detailed-summary.component.tsx [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+                        <div
+                          className={`${styles.centerItems} ${styles.alignLeft}`}
+                        >
+                          <span>
+                            {capitalize(condition.resource.clinicalStatus)}
+                          </span>
+                        </div>
+                      </td>
+                      <td>
+                        <svg className="omrs-icon" fill="rgba(0, 0, 0, 0.54)">
```
Comment (donaldkibet) [inline src/widgets/conditions/conditions-detailed-summary.component.tsx]:
```suggestion
                        <svg className="omrs-icon" fill="var(--omrs-color-ink-low-contrast)">
```
recommend we use the style guide colors [style guide](https://styleguide.openmrs.org/?path=/story/openmrs-styleguide--colors)

File: translations/en.json (hunk 2 above)
Comment (denniskigen) [reply translations/en.json]: Ok.

File: src/widgets/conditions/conditions-detailed-summary.component.test.tsx (hunk 1 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.component.test.tsx]: Ok

File: src/widgets/conditions/conditions-detailed-summary.component.tsx (hunk 3 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.component.tsx]: Yes, it should. Amended.

File: src/widgets/conditions/conditions-detailed-summary.css (hunk 5 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.css]: That background property shouldn't be there. I was using it for testing. I've removed it.

File: src/widgets/conditions/conditions-detailed-summary.component.test.tsx (hunk 1 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.component.test.tsx]: Yes

File: src/widgets/conditions/conditions-detailed-summary.component.tsx (hunk 3 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.component.tsx]: Agree. Fixed.

File: src/widgets/conditions/conditions-detailed-summary.css (hunk 4 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.css]: Done

File: src/widgets/conditions/conditions-detailed-summary.css (hunk 6 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.css]: Done

File: src/widgets/conditions/conditions-detailed-summary.component.test.tsx (hunk 8 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.component.test.tsx]: Fixed

File: src/widgets/conditions/conditions-detailed-summary.component.tsx (hunk 9 above)
Comment (denniskigen) [reply src/widgets/conditions/conditions-detailed-summary.component.tsx]: Done

**Pull request merged**